r2/service.go

```diff
@@ -59,113 +78,569 @@ func NewService(iOpts instrument.Options, rootPrefix string, store Store) server
 	return &service{iOpts: iOpts, store: store, rootPrefix: rootPrefix}
 }
 
-func (s *service) readNamespaces(w http.ResponseWriter, _ *http.Request) {
-	w.WriteHeader(http.StatusNotImplemented)
-	fmt.Fprintf(w, "Placeholder for readNamespace")
+type payload struct {
```

Comment:
I don't understand the use of this type. Why do you need to indicate the status code here? Won't the HTTP layer address that?